test/NuGet.Core.FuncTests/NuGet.XPlat.FuncTest/XPlatHelpOutputTests.cs

```diff
@@ -82,5 +101,32 @@ public void MainInternal_OnParsingSuccess_DoesNotShowsHelp(string[] args)
             Assert.DoesNotContain("Usage", output);
             Assert.Equal(1, exitCode);
         }
+
+        [Fact]
+        public void MainInternal_ShowsHelp()
+        {
+            // Arrange
+            var originalConsoleOut = Console.Out;
+            using var consoleOutput = new StringWriter();
+            Console.SetOut(consoleOutput);
+            var log = new TestCommandOutputLogger(_testOutputHelper);
+
+            // Act
+            var exitCode = Program.MainInternal(Array.Empty<string>(), log);
+            Console.SetOut(originalConsoleOut);
+
+            // Assert
+            var output = consoleOutput.ToString();
+
+            var commandPattern = @"^\s{2}(\w+)\s{2,}"; // Matches lines starting with two spaces, a word (command), followed by at least two spaces
+            IEnumerable<string> matches = Regex.Matches(output, commandPattern, RegexOptions.Multiline).Select(m => m.ToString().Trim());
+
+            foreach (var command in HelpCommands)
+            {
+                Assert.Contains(command, matches);
+            }
```

Comment:
This seems to O(nsquare) time complexity. Is there any other API that compares the collection for equality in xUnit? Maybe `Assert.Equal(HelpCommands, matches.OrderBy(m => m));`